PG-1366 Clean up architecture documentation and make it up to date

jeltz opened the pull request:
The architecture documentation was outdated so this makes it up to date plus improves various minor issues found while updating the documentation.

Some function names are fixed in #229.

jeltz committed:
PG-1366 Clean up architecture documentation and make it up to date

The architecture documentation was outdated so this makes it up to date
plus improves various minor issues found while updating the
documentation.

AndersAstrand commented on contrib/pg_tde/documentation/docs/architecture.md:
```diff
@@ -44,35 +45,30 @@ In the future these could be extracted into separate shared libraries with an op
 `pg_tde` currently uses the following encryption algorithms:
 
-* `AES-CBC-128` algorithm for encrypting most database files.
+* `AES-128-CBC` for encrypting database files; encrypted with intenral keys.
```
s/intenral/internal/

AndersAstrand commented on contrib/pg_tde/documentation/docs/architecture.md:
```diff
@@ -44,35 +45,30 @@ In the future these could be extracted into separate shared libraries with an op
+* `AES-128-CBC` for encrypting database files; encrypted with intenral keys.
 
-    Here's how it works:
+* `AES-128-CTR` for WAL encryption; encrypted with intenral keys.
```
s/intenral/internal/

AndersAstrand commented on contrib/pg_tde/documentation/docs/architecture.md:
```diff
@@ -84,18 +80,19 @@ The `tde_heap` access method is the same as the `heap` one. It uses the same fun
-WAL encryption is controlled globally via a global GUC variable that requires a server restart.
+WAL encryption is controlled globally via a global GUC variable, `pg_tde.wal_encrypt`, that requires a server restart.
+
+WAL keys also contain the [LSN](https://www.postgresql.org/docs/17/wal-internals.html) of the first WAL write after key creation. This allows to know what WAL ranges are encrypted (or not) with which key.
```
This allows _who_ to know?

jeltz replied:
Fixed this.

AndersAstrand commented on contrib/pg_tde/documentation/docs/architecture.md:
```diff
@@ -294,9 +279,7 @@ pg_tde_(grant/revoke)_local_provider_management_to_role
-There's one special behavior:
-When `pg_tde.inherit_global_providers` is ON, users with database local permissions can list global providers, but can't use the show function to query configuration details.
-When `pg_tde.inherit_global_providers` is OFF, they can't execute the function at all, it will return an error.
+There's one special behavior: When `pg_tde.inherit_global_providers` is `on`, users with database local permissions can list global providers, but can't use the show function to query configuration details. When `pg_tde.inherit_global_providers` is `off`, they can't execute the function at all, it will return an error.
```
Which show function?

jeltz replied:
No idea, probably something which does not exist anymore.

jeltz replied:
Removed it.

jeltz committed:
PG-1366 Remove too detailed and incorrect comment from achitecture docs

The fucntions for lsiting keys do not look at inherit_global_providers
but even if they did it does not seem like something which would belong
in the architecture documentation.

Andriciuc committed:
Update a small fix to sequences

Fixed sequences from seqeunces

it-percona-cla commented:
[![CLA assistant check](https://cla-assistant.percona.com:443/pull/badge/not_signed)](https://cla-assistant.percona.com:443/percona/postgres?pullRequest=246) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.percona.com:443/percona/postgres?pullRequest=246) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: jeltz<br/>:x: Andriciuc<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.percona.com:443/check/percona/postgres?pullRequest=246) it.</sub>

Andriciuc reviewed (approved):
Reviewed and approved for merging. Excellent update to the processes and overall architecture description for tde, thank you!
